lib/workflow/stores/mongo.js

```diff
@@ -52,22 +52,26 @@ function mongoStoreFactory(waterline, Promise, Constants, Errors, assert, _) {
      * @param {Object} indexObject an object with keys that correspond to the mogo
      * collections on which to place indexes and values that are arrays of indexes
      * @example {
-     *          taskdependencies: [
-     *              {taskId: 1},
-     *              {graphId: 1},
-     *              {taskId: 1, graphId: 1}
-     *          ],
-     *          graphobjects: [
-     *              {instanceId: 1}
-     *          ]
+     *       taskdependencies: [
+     *           {index: {taskId: 1}, unique: true},
```

Comment:
Would it make more sense to keep this type of information in the model?  Is this really a mongo only behavior?

If all databases will need taskId to be unique, then the waterline 'unique' attribute could be set to 'true' in the model.  If all databases should index this key/row, then the waterline 'index' attribute could be set to 'true'.